cmd/kube-proxy/app/server_others.go

```diff
@@ -145,6 +152,38 @@ func newProxyServer(
 		return nil, fmt.Errorf("cannot determine detect-local-mode: %v", err)
 	}
 
+	var nodeInfo *v1.Node
+	if detectLocalMode == proxyconfigapi.LocalModeNodeCIDR {
+		klog.Infof("Watching for node %s, awaiting podCIDR allocation", hostname)
+
+		// since allocators can assign the podCIDR after the node registers, we do a watch here to wait
+		// for podCIDR to be assigned, instead of assuming that the Get() on startup will have it.
+		w, err := client.CoreV1().Nodes().Watch(context.TODO(), metav1.ListOptions{FieldSelector: fmt.Sprintf("metadata.name=%s", hostname)})
```

Comment:
is this a reliable way of finding "our" node?
it seems that some cloud-providers doesn't respect it
https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#built-in-node-labels

> Note: The value of these labels is cloud provider specific and is not guaranteed to be reliable. For example, the value of kubernetes.io/hostname may be the same as the Node name in some environments and a different value in other environments.

another option is look for the nodeIP in Status, that is obtained just before this logic
https://github.com/kubernetes/kubernetes/pull/88935/files#diff-934760ea1a4e195e49bcd10625cf4e46L148

```
  status:
    addresses:
    - address: 172.17.0.3
      type: InternalIP
    - address: kind-worker
      type: Hostname
```

Reply:
I would simply go with node name. kube-proxy should already have the right hostname (which needs to match the node name) or the admin can use the --hostname-override option to set the right node name. IP addresses are more complicated because in dual-stack there can be multiple addresses, plus there are multiple Node object addresses to look through. But there's only one node name, and that has to be correct anyway.

Reply:
isn't that what matadata.name is doing? basically going by the node name? and if you see above in the code, the nodeRef object is also using the hostname. There is no other references to node name that I saw. Am I missing something?

Reply:
metadata.name should be correct - see also cmd/kube-proxy/app/server.go